src/utils/deployments.cairo

```diff
@@ -0,0 +1,89 @@
+mod interface;
+
+use core::pedersen::PedersenTrait;
+use hash::{HashStateTrait, HashStateExTrait};
+use interface::IUniversalDeployer;
+use openzeppelin::utils::serde::SerializedAppend;
+use poseidon::PoseidonTrait;
+use starknet::ClassHash;
+use starknet::ContractAddress;
+
+// 2**251 - 256
+const L2_ADDRESS_UPPER_BOUND: felt252 =
+    0x7ffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff00;
+const CONTRACT_ADDRESS_PREFIX: felt252 = 'STARKNET_CONTRACT_ADDRESS';
+
+/// Returns the contract address from a `deploy_syscall`.
+/// `deployer_address` should be the zero address if the deployment is origin-independent (deployed from zero).
+/// For more information, see https://docs.starknet.io/documentation/architecture_and_concepts/Smart_Contracts/contract-address/
+fn calculate_contract_address_from_deploy_syscall(
```

Comment:
Small last comment, sorry since I was the one suggesting the change, but what do you think of putting calculate_contract_address at the beginning of the helper name? I think it feels better that way, since the most important part is that, no the deploy_syscall.

TL'DR: I vote we change deploy_syscall_calculate_contract_address to calculate_contract_address_from_deploy_syscall, and udc_calculate_contract_address to calculate_contract_address_from_udc

Reply:
Haha no worries. Yeah, it seems more natural starting with `calculate_`. Will update